File: LeetCode/medium/Unique_Paths.cxx
class Solution {
public:
    int uniquePaths(int M, int N) {
        long long m = std::max(M, N) - 1;
        long long n = std::min(M, N) - 1;

        unsigned long long nm = 1;
        for (int i = m+1; i <= n+m; ++i) nm *= i;
        
        unsigned long long _n = 1;
        for (int i = 1; i <= n; ++i) _n *= i;
        
         
        return nm / _n;
    }
};

File: LeetCode/medium/Unique_Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        def fact(x):
            if x <= 1:
                return 1
            else:
                return x * fact(x-1)
        m -= 1
        n -= 1
        n += m     
        return int(fact(n) / (fact(m) * fact(n-m)))
